src/main/java/org/broadinstitute/hellbender/cmdline/CommandLineProgram.java

```diff
@@ -165,8 +166,44 @@ public Object instanceMainPostParseArgs() {
 
         // set the temp directory as a java property, checking for existence and read/write access
         final Path p = tmpDir.toPath();
+        Path tempFilePath = null;
         try {
             p.getFileSystem().provider().checkAccess(p, AccessMode.READ, AccessMode.WRITE);
+
+            // Warn if there's anything that prevents execution in the tmp dir because some tools need that
+            // This test relies on the file system supporting posix file permissions
+            if(p.getFileSystem().supportedFileAttributeViews().contains("posix")) {
```

Comment:
I'll move this into a separate function but I'm not sure there's a great way to fix the nested try / catch issue.  I want the inner one because it specifically catches issues setting exec permissions on the temp file and I don't think that function throws granular-enough exceptions for me to easily be able to distinguish between an exception from that and an exception from like creating the file or getting the posix permissions

Reply:
I see your point and it makes sense.  I'm not sure we care that much about treating those cases differently though?  Can we give meaningfully different advice if getting permissions failed vs if setting exec failed?  Maybe can.  If you think that's a useful distinction I'm not against having it split out.  I do like it all encapsulated in a function that won't explode in any case though.

Reply:
Yeah, I think maybe it's not a worthwhile distinction